packages/contractkit/src/providers/celo-private-keys-subprovider.ts

```diff
@@ -96,13 +110,13 @@ export class CeloPrivateKeysWalletProvider extends PrivateKeyWalletSubprovider {
       txParams.chainId = await this.getChainId()
     }
 
-    if (txParams.nonce == null) {
+    if (txParams.chainId == null) {
```

Comment:
bug? check the chainInd, set the nonce?

Reply:
which makes me think we need through full unit test for this logic. It's really core

Reply:
> bug? check the chainInd, set the nonce?

Yeah, copy-pate bug.

Reply:
> which makes me think we need through full unit test for this logic. It's really core

I agree

Reply:
Add all possible permutations to `tx-signing.test.ts`